src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,163 @@
+        Node<T> node = getNodeByIndex(index);
+        T oldValue = node.value;
+        node.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        T removedValue;
+        if (index == 0) {
+            removedValue = head.value;
+            head = head.next;
+            if (head == null) {
+                tail = null;
+            }
+        } else {
+            Node<T> prev = getNodeByIndex(index - 1);
+            removedValue = prev.next.value;
+            prev.next = prev.next.next;
+            if (index == size - 1) {
+                tail = prev;
+            }
+        }
+        size--;
+        return removedValue;
     }
 
     @Override
     public boolean remove(T object) {
+        if (head == null) {
+            return false;
+        }
+
+        if (object == null) {
+            if (head.value == null) {
+                remove(0);
+                return true;
+            }
+        } else if (object.equals(head.value)) {
+            remove(0);
+            return true;
+        }
+
+        Node<T> current = head;
+        while (current.next != null) {
+            if (current.next.value == null && object == null) {
+                current.next = current.next.next;
+                if (current.next == null) {
+                    tail = current;
+                }
+                size--;
+                return true;
+            } else if (current.next.value != null && current.next.value.equals(object)) {
+                current.next = current.next.next;
+                if (current.next == null) {
+                    tail = current;
+                }
+                size--;
+                return true;
+            }
+            current = current.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void checkIndexForAdd(int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Invalid index: "
+                    + index + ". Valid indices are between 0 and " + (size)
+                    + ", list size is " + size);
+        }
+    }
+
+    private void checkIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Invalid index: "
+                    + index + ". Valid indices are between 0 and " + (size - 1));
+        }
+    }
+
+    private Node<T> getNodeByIndex(int index) {
+        Node<T> current = head;
+        for (int i = 0; i < index; i++) {
+            current = current.next;
+        }
+        return current;
+    }
+
+    private class Node<T> {
+        private T value;
+        private Node<T> next;
+
+        public Node(T value) {
+            this.value = value;
+        }
```

Comment:
Make `Node<T>` an inner class at the end of the `MyLinkedList` class. Also, don't use getters and setters in class Node; directly access the private fields of the inner class.